ocaml/xapi/helpers.ml

```diff
@@ -2282,13 +2282,24 @@ module AuthenticationCache = struct
 
     type session = Secret.secret
 
-    type t = {cache: Q.t; mutex: Mutex.t; elapsed: Mtime_clock.counter}
+    type t = {
+        cache: Q.t
+      ; mutex: Mutex.t
+      ; elapsed: Mtime_clock.counter
+            (* Counter that can be queried to
+               find out how much time has elapsed since the cache's
+               construction. This is used as a reference point when creating and
+               comparing expiration spans on cache entries. *)
+      ; expiry: Mtime.span
+            (* Time-to-live assigned to each cache entry upon insertion. *)
```

Comment:
Using expiry as a name sounds a bit strange, I'd use ttl as short for time to leave, along with a different explanation. Feel free to modify or expand it:
```ocaml
      ; ttl: Mtime.span
            (* Time-to-live: amount of time that a cache entry is valid. After that time the entry expires. *)
```